go/batch-submitter/batch-tx/batch.go

```diff
@@ -0,0 +1,185 @@
+package batchtx
+
+import (
+	"errors"
+
+	l2types "github.com/ethereum-optimism/optimism/l2geth/core/types"
+)
+
+var (
+	// ErrBlockWithInvalidContext signals an attempt to generate a
+	// BatchContext that specifies a total of zero txs.
+	ErrBlockWithInvalidContext = errors.New("attempted to generate batch " +
+		"context with 0 queued and 0 sequenced txs")
+)
+
+// BatchElement reflects the contents of an atomic update to the L2 state.
+// Currently, each BatchElement is constructed from a single block containing
+// exactly one tx.
+type BatchElement struct {
+	// Timestamp is the L1 timestamp of the batch.
+	Timestamp uint64
+
+	// BlockNumber is the L1 BlockNumber of the batch.
+	BlockNumber uint64
+
+	// Tx is the optional transaction that was applied in this batch.
+	//
+	// NOTE: This field will only be populated for sequencer txs.
+	Tx *l2types.Transaction
+}
+
+// IsSequencerTx returns true if this batch contains a tx that needs to be
+// posted to the L1 CTC contract.
+func (b *BatchElement) IsSequencerTx() bool {
+	return b.Tx != nil
+}
+
+// BatchElementFromBlock constructs a BatchElement from a single L2 block. This
+// method expects that there is exactly ONE tx per block. The returned
+// BatchElement will reflect whether or not the lone tx is a sequencer tx or a
+// queued tx.
+func BatchElementFromBlock(block *l2types.Block) BatchElement {
+	tx := block.Transactions()[0]
```

Comment:
We should check to make sure there is a tx in the block to prevent a panic here, the genesis block won't have a tx in it

Reply:
afaict the genesis block is explicitly skipped by setting `BLOCK_OFFSET=1`, so we should never process it?